File: s-recursos/src/main/java/co/edu/javeriana/arqui/rest/ejb/ServicioClinicas.java
package co.edu.javeriana.arqui.rest.ejb;

import co.edu.javeriana.arqui.rest.entity.Clinica;
import co.edu.javeriana.arqui.rest.repository.RepositorioClinicas;
import jakarta.ejb.EJB;
import jakarta.ejb.Stateless;
import java.util.List;

@Stateless
public class ServicioClinicas {

    @EJB
    private RepositorioClinicas repoClinicas;

    public Clinica registrarClinica(Clinica clinica) {
        repoClinicas.crear(clinica);
        return clinica;
    }

    public List<Clinica> listarClinicas() {
        return repoClinicas.listarTodos();
    }
}

File: s-recursos/src/main/java/co/edu/javeriana/arqui/rest/ejb/ServicioClinicas.kt
package co.edu.javeriana.arqui.rest.ejb

import co.edu.javeriana.arqui.rest.entity.Clinica
import co.edu.javeriana.arqui.rest.repository.RepositorioClinicas
import jakarta.ejb.EJB
import jakarta.ejb.Stateless


@Stateless
open class ServicioClinicas {

    @EJB
    private lateinit var repoClinicas: RepositorioClinicas

    fun registrarClinica(clinica: Clinica): Clinica {
        repoClinicas.crear(clinica)
        return clinica
    }

    fun listarClinicas(): List<Clinica> {
        return repoClinicas.listarTodos()
    }
}